controllers/fluxdeployer_controller.go

```diff
@@ -151,24 +152,28 @@ func (r *FluxDeployerReconciler) reconcile(ctx context.Context, obj *v1alpha1.Fl
 
 	snapshotURL := fmt.Sprintf("oci://%s/%s", r.RegistryServiceName, snapshotRepo)
 
-	// create oci registry
-	if err := r.reconcileOCIRepo(ctx, obj, snapshotURL, snapshot.Spec.Tag); err != nil {
-		msg := "failed to create or update oci repository"
-		logger.Error(err, msg)
-		conditions.MarkFalse(obj, meta.ReadyCondition, v1alpha1.CreateOrUpdateOCIRepositoryFailedReason, err.Error())
-		conditions.MarkStalled(obj, v1alpha1.CreateOrUpdateOCIRepositoryFailedReason, err.Error())
-		event.New(r.EventRecorder, obj, eventv1.EventSeverityError, msg, nil)
-		return ctrl.Result{}, err
+	// create kustomization
+	if obj.Spec.KustomizationTemplate != nil {
+		// create oci registry
+		if err := r.createKustomizationSources(ctx, obj, snapshotURL, snapshot.Spec.Tag); err != nil {
+			msg := "failed to create kustomization sources"
+			logger.Error(err, msg)
+			conditions.MarkFalse(obj, meta.ReadyCondition, v1alpha1.CreateOrUpdateKustomizationFailedReason, err.Error())
+			conditions.MarkStalled(obj, v1alpha1.CreateOrUpdateKustomizationFailedReason, err.Error())
+			event.New(r.EventRecorder, obj, eventv1.EventSeverityError, msg, nil)
+			return ctrl.Result{}, err
+		}
 	}
 
-	// create kustomization
-	if err = r.reconcileKustomization(ctx, obj); err != nil {
-		msg := "failed to create or update kustomization"
-		logger.Error(err, msg)
-		conditions.MarkFalse(obj, meta.ReadyCondition, v1alpha1.CreateOrUpdateOCIRepositoryFailedReason, err.Error())
-		conditions.MarkStalled(obj, v1alpha1.CreateOrUpdateOCIRepositoryFailedReason, err.Error())
-		event.New(r.EventRecorder, obj, eventv1.EventSeverityError, msg, nil)
-		return ctrl.Result{}, err
+	if obj.Spec.HelmReleaseTemplate != nil {
```

Comment:
this should be an `else` case I believe as we cannot instantiate both.

Reply:
Technically, because we can't instantiate both and we will have a check, this doesn't matter here, right? I think we can have an exclusive or with kubebuilder marks. Or something similar. So both can't be defined together.

Reply:
Ah dang it, so markers will not work. I thought there is something, but no: https://github.com/kubernetes-sigs/controller-tools/issues/461

It hasn't been implemente.

Reply:
i think `reconcile` is a good place yes.